pkg/systemd/services/services.jsx

```diff
@@ -257,6 +257,25 @@ class ServicesPage extends React.Component {
         if ((nextState.loadingUnits === true && this.state.loadingUnits === true) ||
            (this.seenPaths.size == 0 || this.seenPaths.size > Object.keys(nextState.unit_by_path).length))
             return false;
+
+        /* Throttle render rate to 10Hz.  Things like "udevadm
```

Comment:
Since I mentioned on IRC the throttle-debounce npm lib that we use elsewhere as well:
I think its a bit cleaner but I am all good with this as well :)

So with that it would be something like:

in
```
shouldComponentUpdate() {
    debounce(100, () => {
        this.render_holdoff = false;
        this.setState({});
    })
    return !this.render_holdoff;
}
```
And in:
```
componentDidUpdate() {
   this.render_holdoff = true;
}
```
And of course this.render_holdoff = true should be also in the constructor.

I did not try this locally.

Reply:
Calling `debounce(100, ...)` doesn't do anything, right?  Once needs to at least call the returned wrapper: `debounce(100, ...)()`, but then this is just an obscure way to call window.setTimeout, which at least has decent documentation.

Reply:
You can cancel debounced call easily, by calling cancel() on the returned value of the debounce. That's an important feature I think, because one you rendered you want to cancel all future triggered debounces.

But as I said - if this will work - I am happy with this as well.

Reply:
> because once you rendered you want to cancel all future triggered debounces.

Yeah, we need to cancel the timer when we render immediately, such as with the currentFilterText changes.  Also when the component is unmounted.

Reply:
Done.